### DISCUSSION TITLE
CheckboxGroup default value from state

### DISCUSSION THREAD

**Discussion by Arthur-Ruthenberg** [chakra-ui/chakra-ui#5860] [Q&A]
Hi experts, 
I am using Chakra with React Typescript and implementing a checkbox group The default values are controlled by an outside state that is passed down as a prop.

The problem is that the CheckboxGroup doesn't accept the default values from outside source

The code is as follows:

```
import React, {FC, useCallback, useEffect, useState} from "react";
import { CheckboxGroup, Checkbox, VStack } from "@chakra-ui/react";

interface IGroupCheckbox {
    values: StringOrNumber[],
    labels: StringOrNumber[],
    activeValues: StringOrNumber[]
    onChange: (value:StringOrNumber[])=> void

}
const GroupCheckbox:FC<IGroupCheckbox> = ({
    values,
    labels,
    activeValues,
    onChange
}) => {
    const [currActiveValues, setCurrActiveValues] = useState<StringOrNumber[]>();
    const handleChange = useCallback((value:StringOrNumber[]) => {
        if(value?.length === 0) {
            alert('you must have at least one supported language');
            return;
        }
        onChange(value);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    useEffect(()=>{
        if(activeValues) {
            setCurrActiveValues(['en'])
        }
    },[activeValues])

    return (
        <CheckboxGroup 
            onChange={handleChange}
            defaultValue={currActiveValues}
        >
            <VStack>
                {values && labels && values.map((item:StringOrNumber, index:number)=>
                    {
                        return (
                            <Checkbox 
                                key={item} 
                                value={item}
                            >
                                {labels[index]}
                            </Checkbox>
                        )
                    }
                )}
            </VStack>
        </CheckboxGroup>
    )
}
```

export default GroupCheckbox
When I change the defaultValue parameter, instead of the state managed, to be `defaultValue={['en']} `it works fine, but any other input for this prop doesn't work. I checked and triple checked that the values are correct.
Any suggestions?

**Comment by eckertalex** [comment]
I am not entirely sure why, but it seems to be connected with the `useEffect`. I don't see a reason why we need the `useEffect` anyway, so I would just remove it. Without knowing how you use the the `GroupCheckbox` I came up with this code sandbox. There it does work.

https://codesandbox.io/s/chakra-discussion-checkboxgroup-5860-hct0xs?file=%2Fsrc%2Fapp.tsx

**Reply by Arthur-Ruthenberg** [reply]
Thanks
I was told that  my error is in using the defaultValue, as it is considered only on first render. 
I haven't had a chance to check it yet, but defaultValue is actually considered on first render only, it is logical that this what causes my problem.
